deep-learning/_posts/2018-07-27-SFCNN.md

```diff
@@ -0,0 +1,86 @@
+---
+layout: review
+title: Rotation Equivariant CNNs
+tags: deep-learning CNN essentials
+author: "Daniel Jörgens"
+cite:
+    authors: "Weiler, M and Hamprecht, F A and Storath, M"
+    title:   "Learning Steerable Filters for Rotation Equivariant CNNs"
+    venue:   "CVPR 2018"
+pdf: "https://arxiv.org/pdf/1711.07289.pdf"
+---
+
+
+# Contribution
+
+Observing that regular CNNs exhibit translational invariance but suffer from rotational variance, the authors
+propose a CNN architecture based on steerable filters (called *SFCNN*) featuring rotation equivariance. Employing their approach
+circumvents the need for rotation in data augmentation and reduces the '*sample complexity*' of the network.
+The authors' main argument in support of equivariance properties of a network is that by this, the hypothesis
+space (i.e. the possible transformations found in the data) is reduced which simplifies the learning task. 
+
+<p style="text-align:center"><img src="/deep-learning/images/SFCNN/overview.png" width="700"></p>
+
+
+# Steerable Filters
+
+ * A filter $$\Psi$$ is rotationally steerable if "*its rotation by an arbitrary angle $$\theta$$ can
+   be expressed*" in terms of rotations applied to a set of atomic basis functions
+   $$\{\psi_q\}^Q_{q=1}$$ resulting in
+   
+     $$\rho_\theta\Psi(x) = \sum^Q_{q=1} \kappa_q(\theta)\psi_q(x)$$.
+     
+ * The authors choose circular harmonics as the atomic basis functions:
+     
+     $$\psi_q \equiv \psi_{jk} = \tau_j(r) e^{\mathrm{i}k\phi}$$.
+     
+ * A specific learned filter is obtained by linear combination of these basis functions using weights
+   $$w_{jk} \in \mathcal{C}$$ ($$\mathcal{C}$$ denotes the complex numbers ;) ) learned by the network:
+     
+     $$\tilde{\Psi}(x) = \sum^J_{j=1} \sum^{K_j}_{k=0} w_{jk}\psi_{jk}(x)$$.
+ 
+   By choosing the learned weights to be complex, they induce rotations of the basis functions
```

Comment:
May be more clear to add :  "and sharing the real values in between rotated versions of a same filter "

Reply:
Sorry, I do not understand your point here. Can you explain? :)

Reply:
To obtain the same filter rotated, I believe the main idea was to use complex expressions because then you can express rotation as a phase shift. Hence the real parts of the filters are shared? But maybe it isn't why they use complex?

Reply:
In my understanding, they learn the weights w_{jk} which then define the different filters by linear combination. I understand their comment on the complex nature of these weights as to point out that each of those weights actually consists of two values and interpreting them as a complex number causes (relative) rotations of the basis filters to be learned. I did not see a comment where they explicitly talk about sharing these weights. Or did I miss that?